components/Modal/TokenAuctionBidModal.js

```diff
@@ -0,0 +1,295 @@
+import { useForm } from 'react-hook-form'
+import useStore from 'lib/store'
+import { prettyBalance } from 'utils/common'
+import { useIntl } from 'hooks/useIntl'
+import Modal from 'components/Common/Modal'
+import Button from 'components/Common/Button'
+import { InputText } from 'components/Common/form'
+import { sentryCaptureException } from 'lib/sentry'
+import { GAS_FEE, STORAGE_ADD_MARKET_FEE } from 'config/constants'
+import { formatNearAmount, parseNearAmount } from 'near-api-js/lib/utils/format'
+import JSBI from 'jsbi'
+import { IconX } from 'components/Icons'
+import { useEffect, useState } from 'react'
+import useProfileData from 'hooks/useProfileData'
+import { flagColor, flagText } from 'constants/flag'
+import BannedConfirmModal from './BannedConfirmModal'
+import WalletHelper from 'lib/WalletHelper'
+import { trackUpdateListingToken } from 'lib/ga'
+import { useToast } from 'hooks/useToast'
+
+const TokenAuctionBidModal = ({
+	data,
+	show,
+	onClose,
+	bidAuctionAmount,
+	onSuccess,
+	tokenType = `token`,
+}) => {
+	const [showBannedConfirm, setShowBannedConfirm] = useState(false)
+	const creatorData = useProfileData(data?.metadata.creator_id)
+	const { localeLn } = useIntl()
+	const { errors, register, handleSubmit, watch } = useForm({
+		defaultValues: {
+			bidAuctionAmount,
+		},
+	})
+	const [isBidding, setIsBidding] = useState(false)
+	const { currentUser, userBalance, setTransactionRes } = useStore((state) => ({
+		currentUser: state.currentUser,
+		userBalance: state.userBalance,
+		setTransactionRes: state.setTransactionRes,
+	}))
+	const toast = useToast()
+
+	useEffect(() => {
+		checkNextPriceBid()
+	}, [])
+
+	const hasStorageBalance = async () => {
+		try {
+			const currentStorage = await WalletHelper.viewFunction({
+				methodName: 'storage_balance_of',
+				contractId: process.env.MARKETPLACE_CONTRACT_ID,
+				args: { account_id: currentUser },
+			})
+
+			const supplyPerOwner = await WalletHelper.viewFunction({
+				methodName: 'get_supply_by_owner_id',
+				contractId: process.env.MARKETPLACE_CONTRACT_ID,
+				args: { account_id: currentUser },
+			})
+
+			const usedStorage = JSBI.multiply(
+				JSBI.BigInt(parseInt(supplyPerOwner) + 1),
+				JSBI.BigInt(STORAGE_ADD_MARKET_FEE)
+			)
+
+			if (JSBI.greaterThanOrEqual(JSBI.BigInt(currentStorage), usedStorage)) {
+				return true
+			}
+			return false
+		} catch (err) {
+			sentryCaptureException(err)
+		}
+	}
+
+	const onPlaceBidAuction = async ({ bidAuctionAmount }) => {
+		setIsBidding(true)
+
+		trackUpdateListingToken(data.token_id)
+
+		const hasDepositStorage = await hasStorageBalance()
+
+		try {
+			const depositParams = { receiver_id: currentUser }
+
+			const params = {
+				nft_contract_id: data.contract_id,
+				...(data.token_id
+					? { token_id: data.token_id }
+					: { token_series_id: data.token_series_id }),
+				ft_token_id: 'near',
+				amount: parseNearAmount(bidAuctionAmount),
+			}
+
```

Comment:
same comment as above, tracker and params

Reply:
wokaay hu